### PR TITLE
Add forms in view login, signup.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

- [x] Add new views login, sigunp, with `yiisoft/form`. 
- [x] Add tests `LoginAcceptance` and `SignupAcceptance`.

image

image

### CODE REVIEW

File: views/auth/login.php [hunk 1]
```diff
@@ -2,51 +2,58 @@
...
+    <div class="row d-flex justify-content-center align-items-center h-100">
+        <div class="col-12 col-md-8 col-lg-6 col-xl-5">
+            <div class="card border border-dark shadow-2-strong rounded-3">
+                <div class="card-header bg-dark text-white">
+                    <h1 class="fw-normal h3 text-center"><?= Html::encode($this->getTitle()) ?></h1>
+                </div>
+                <div class="card-body p-5 text-center">
+                    <?= Form::widget()
+                        ->action($urlGenerator->generate('auth/login'))
+                        ->attributes(['enctype' => 'multipart/form-data', 'novalidate' => true])
```
Comment (samdark) [inline views/auth/login.php:33]: Why novalidate?

File: views/layout/main.php [hunk 2]
```diff
@@ -36,99 +37,141 @@
...
-<?php
-$this->beginBody();
+?>
+<!DOCTYPE html>
+<html class="h-100" lang="en">
+    <head>
+        <meta charset="utf-8">
+        <meta http-equiv="X-UA-Compatible" content="IE=edge">
+        <meta name="viewport" content="width=device-width, initial-scale=1">
+        <title>Yii Demo<?= $this->getTitle() ? ' - ' . $this->getTitle() : '' ?></title>
```
Comment (samdark) [inline views/layout/main.php:47]:
```suggestion
        <title>Yii Demo<?= $this->getTitle() ? ' - ' . Html::encode($this->getTitle()) : '' ?></title>
```

File: views/signup/signup.php [hunk 3]
```diff
@@ -2,42 +2,51 @@
...
  * @var \Yiisoft\View\WebView $this
  * @var \Yiisoft\Translator\TranslatorInterface $translator
  * @var \Yiisoft\Router\UrlGeneratorInterface $urlGenerator
  * @var string $csrf
+ * @var FormModelInterface $formModel
  */
 
-$this->setTitle(Html::encode($translator->translate('signup')));
-$error = $error ?? null;
+$this->setTitle(Html::encode($translator->translate('Signup')));
```
Comment (samdark) [inline views/signup/signup.php:18]: Should not encode.

File: views/signup/signup.php [hunk 4]
```diff
@@ -2,42 +2,51 @@
...
+    <div class="row d-flex justify-content-center align-items-center h-100">
+        <div class="col-12 col-md-8 col-lg-6 col-xl-5">
+            <div class="card border border-dark shadow-2-strong rounded-3">
+                <div class="card-header bg-dark text-white">
+                    <h1 class="fw-normal h3 text-center"><?= Html::encode($this->getTitle()) ?></h1>
+                </div>
+                <div class="card-body p-5 text-center">
+                    <?= Form::widget()
+                        ->action($urlGenerator->generate('auth/signup'))
+                        ->attributes(['enctype' => 'multipart/form-data', 'novalidate' => true])
```
Comment (samdark) [inline views/signup/signup.php:31]: Why novalidate?

File: views/layout/main.php (hunk 2 above)
Comment (terabytesoftw) [reply views/layout/main.php:47]: done.

File: views/signup/signup.php (hunk 3 above)
Comment (terabytesoftw) [reply views/signup/signup.php:18]: done.

File: views/auth/login.php (hunk 1 above)
Comment (terabytesoftw) [reply views/auth/login.php]: fixed.

File: views/signup/signup.php (hunk 4 above)
Comment (terabytesoftw) [reply views/signup/signup.php]: Fixed.

File: resources/messages/ru/app.php [hunk 5]
```diff
@@ -42,4 +42,7 @@
     'layout.add.random-content' => 'Добавить случайный контент',
     'layout.migrations' => 'Миграции',
     'layout.db.schema' => 'Схема БД',
+    'validator.invalid.login.paswword' => 'Неверный логин или пароль',
```
Comment (vjik) [inline resources/messages/ru/app.php:45]:
```suggestion
    'validator.invalid.login.password' => 'Неверный логин или пароль',
```

File: src/Auth/Form/LoginForm.php [hunk 6]
```diff
@@ -56,7 +56,7 @@ function (): Result {
                 $result = new Result();
 
                 if (!$this->authService->login($this->login, $this->password, $this->rememberMe)) {
-                    $result->addError($this->translator->translate('Invalid login or password'));
+                    $result->addError($this->translator->translate('validator.invalid.login.paswword'));
```
Comment (vjik) [inline src/Auth/Form/LoginForm.php:59]:
```suggestion
                    $result->addError($this->translator->translate('validator.invalid.login.password'));
```

File: views/auth/login.php [hunk 7]
```diff
@@ -2,51 +2,58 @@
...
+                </div>
+                <div class="card-body p-5 text-center">
+                    <?= Form::widget()
+                        ->action($urlGenerator->generate('auth/login'))
+                        ->attributes(['enctype' => 'multipart/form-data'])
+                        ->csrf($csrf)
+                        ->id('loginForm')
+                        ->begin() ?>
+
+                        <?= Field::widget()->config($formModel, 'login')->text(['autofocus' => true]) ?>
```
Comment (vjik) [inline views/auth/login.php:38]: On error login also should be red:

image

**Review by vjik — COMMENTED** [review]
The situation when the user not exists identity is not handled. On login throws exception `App\User\User::getIdentity(): Return value must be of type App\Auth\Identity, null returned`

File: resources/messages/ru/app.php (hunk 5 above)
Comment (terabytesoftw) [reply resources/messages/ru/app.php:45]: done.

File: src/Auth/Form/LoginForm.php (hunk 6 above)
Comment (terabytesoftw) [reply src/Auth/Form/LoginForm.php:59]: done.

File: views/auth/login.php (hunk 7 above)
Comment (terabytesoftw) [reply views/auth/login.php]: fixed.

**Review by vjik — APPROVED** [review]

**Comment by terabytesoftw** [comment]
`Contact view:`

image

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**